airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -455,17 +462,91 @@ public List<AutoCloseableIterator<AirbyteMessage>> getIncrementalIterators(final
               AutoCloseableIterators.lazyIterator(incrementalIteratorSupplier, null)));
 
     } else if (PostgresUtils.isXmin(sourceConfig) && isIncrementalSyncMode(catalog)) {
-      final XminStateManager xminStateManager = new XminStateManager(stateManager.getRawStateMessages());
-      final PostgresXminHandler handler = new PostgresXminHandler(database, sourceOperations, getQuoteString(), xminStatus, xminStateManager);
-      return handler.getIncrementalIterators(catalog, tableNameToTable, emittedAt);
+      final List<AirbyteStateMessage> rawStateMessages = stateManager.getRawStateMessages();
+
```

Comment:
So as I understand at a very high-level we are : 
1. Identifying streams to sync via ctid
2. Identify streams to sync via xmin
3. Build those interators
4. Concatenate them

If so, can we add a quick comment on this